codec-haproxy/src/main/java/io/netty/handler/codec/haproxy/HAProxyMessage.java

```diff
@@ -519,4 +516,67 @@ public int destinationPort() {
     public List<HAProxyTLV> tlvs() {
         return tlvs;
     }
+
+    @Override
+    public HAProxyMessage touch() {
+        if (leak != null) {
+            leak.record();
+        }
+        return (HAProxyMessage) super.touch();
+    }
+
+    @Override
+    public HAProxyMessage touch(Object hint) {
+        if (leak != null) {
+            leak.record(hint);
+        }
+        return this;
+    }
+
+    @Override
+    public HAProxyMessage retain() {
+        if (leak != null) {
+            leak.record();
+        }
```

Comment:
you can move this to a method and re-use....


```
private void tryRecord() {
    if (leak != null) {
        leak.record();
    }
}
```

Reply:
Done!